### ISSUE TITLE
Cannot undo track's deletion when it was deleted using keyboard only

### ISSUE THREAD

**Issue by dozzzzer** [audacity/audacity#9625]
1. Open a project with a track and a clip
2. Navigate to the track header (until it's highlighted with the black box)
3. Press Enter to select the track
4. Press Delete (Backspace) 
5. Observe: the track has been deleted
6. Undo
7. Observe: the deletion cannot be undone

https://github.com/user-attachments/assets/0c8a8b60-7cc1-4587-bd34-eedc09cbeb3f

**Comment by saintmatthieu** [comment]
Introduced by https://github.com/audacity/audacity/pull/9589

**Comment by igorkorsukov** [comment]
The problem arose because a hack was added in the form of UiCtxProjectPlayback as the current context.
This hack makes it difficult to find out where we are now.
The UI context answers the question of where we are, and not what we can do and what we can’t.

A little about the problem. In audio applications, it's common to activate  Play/Stop via Spacebar. But in MacOS, it's also common to activate controls via Spacebar. All other operating systems via Enter, so the problem is specific to MacOS.
What we have: one shortcut Space and two actions: navigation/trigger and playback/toggle

As I understand it, we want to send playback/toggle if: we are in the project area (project focused?), if the effects dialog is open (or not just open, but also in focus?), if the VST (plugins?) dialog is open, or in focus?
i.e. something like this: send playback/toggle if context Project Focused or EffectsFocused or PluginFocused
Else send navigation/trigger

How to solve the problem?
1. Improve UIAction so that it accepts a combination of contexts, and the context resolver can handle it.
2. Enter syntetic contexts, something like "ProjectFocused_EffectsFocused_PluginFocused" (or even "PlayContext"), BUT it can't be current. It's specified in UIAction, and a context resolver resolves it (MuseScore has one like that - "ProjectNotFocused").
3. Send one action action://activate (global) via Space and some controller processes it and decides what to send next.
